engines/cryomni3d/versailles/documentation.cpp

```diff
@@ -0,0 +1,2070 @@
+			if (!_engine->getCurrentMouseButton()) {
+				// Don't change highlighted date when clicking
+				bool foundBox = false;
+				for (unsigned int i = 0; i < ARRAYSIZE(kTimelineEntries); i++) {
+					if (boxes.hitTest(i, mouse)) {
+						foundBox = true;
+						if (i != hoveredBox) {
+							hoveredBox = i;
+							redraw = true;
+						}
+					}
+				}
+				if (!foundBox && hoveredBox != -1u) {
+					hoveredBox = -1;
+					redraw = true;
+				}
+			}
+			if (_engine->getDragStatus() == kDragStatus_Finished) {
+				if (hoveredBox != -1u) {
+					selectedBox = hoveredBox;
+				}
+				if (boxes.hitTest(leaveBoxId, mouse)) {
+					selectedBox = leaveBoxId;
+				}
+			}
+			if (_engine->checkKeysPressed(1, Common::KEYCODE_ESCAPE)) {
+				selectedBox = leaveBoxId;
+			}
+		}
+		if (g_engine->shouldQuit()) {
+			selectedBox = leaveBoxId;
+		}
+	}
+
+	g_system->showMouse(false);
+
+	delete imageDecoder;
+
+	if (selectedBox == leaveBoxId) {
+		return "";
+	} else {
+		Common::String ret = "VT";
+		ret += kTimelineEntries[selectedBox].year;
+		return ret;
+	}
+}
+
+unsigned int Versailles_Documentation::docAreaHandleRecords(const Common::String &record) {
+	unsigned int action = -1;
+
+	_currentRecord = record;
+	_visitTrace.clear();
+
+	Graphics::ManagedSurface docSurface;
+	Common::String nextRecord;
+	MouseBoxes boxes(10 + ARRAYSIZE(kTimelineEntries));
+
+	while (true) {
+		if (action == -1u) {
+			_currentRecord.toUppercase();
+
+			//debug("Displaying %s", _currentRecord.c_str());
+			docAreaPrepareNavigation();
+			docAreaPrepareRecord(docSurface, boxes);
+			action = docAreaHandleRecord(docSurface, boxes, nextRecord);
+		}
+
+		switch (action) {
+		case 0:
+			action = -1;
+			// Back
+			if (!_visitTrace.empty()) {
+				_currentRecord = _visitTrace.back();
+				_visitTrace.pop_back();
+				break;
+			}
+		// No previous record, like a back to root
+		// fall through
+		case 1:
+			// Back to root
+			return 1;
+		case 2:
+			action = -1;
+			// Follow hyperlink keeping trace
+			_visitTrace.push_back(_currentRecord);
+			_currentRecord = nextRecord;
+			break;
+		case 3:
+			action = -1;
+			// Follow hyperlink losing trace
+			_visitTrace.clear();
+			_currentRecord = nextRecord;
+			break;
+		case 6:
+			// Quit
+			return 2;
+		case 7:
+			action = -1;
+			// General map
+			_visitTrace.clear();
+			nextRecord = docAreaHandleGeneralMap();
+			if (nextRecord == "") {
+				// Go back to current record
+				break;
+			} else if (nextRecord != "VS00") {
+				_currentRecord = nextRecord;
+				break;
+			}
+		// castle has been selected, display its map
+		// fall through
+		case 8:
+			action = -1;
+			// Castle map
+			_visitTrace.clear();
+			nextRecord = docAreaHandleCastleMap();
+			if (nextRecord == "") {
+				// Go back to current record
+				break;
+			} else if (nextRecord != "planG") {
+				_currentRecord = nextRecord;
+				break;
+			} else {
+				// We can't go up to previous case, so let's do a round
+				action = 7;
+				break;
+			}
```

Comment:
I recommend moving `break;` statement out of the ifs to the `case 8:` level, so it does not look like a fall through statement.